underblog.go

```diff
@@ -67,128 +78,138 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
-	t, _ := template.ParseFiles("index.html")
-	t.Execute(f, posts)
+	t, err := template.ParseFiles("index.html")
+	if err != nil {
+		log.Fatalf("Failed to parse: %s", err)
+	}
+	if err := t.Execute(f, posts); err != nil {
+		log.Fatalf("Failed to execute template: %s", err)
+	}
 	f.Close()
 
 	// Copy styles
-	copyCssToPublicDir()
+	if err := copyCSSToPublicDir(); err != nil {
+		log.Fatalf("Failed to copy css: %s", err)
+	}
 
 	elapsed := time.Since(start)
 	log.Printf("Done in %s", elapsed)
 }
 
-func createPost(filename string) Post {
+func createPost(filename string) (*Post, error) {
 	// Get filename without extension
-	filenameBase := FnameWithoutExtension(filename)
-	VerifyFilenameBaseFormat(filenameBase)
+	filenameBase := fnameWithoutExtension(filename)
+	if err := verifyFilenameBaseFormat(filenameBase); err != nil {
+		return nil, err
+	}
 
 	// Get date and slug from filename
 	day := filenameBase[0:2]
 	month := filenameBase[3:5]
 	year := filenameBase[6:10]
 	date, err := time.Parse("02-01-2006", day+"-"+month+"-"+year)
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
 	}
 	slug := filenameBase[11:]
 
 	// Get body from file
 	mdfile, err := os.Open("./markdown/" + filename)
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
 	}
+	defer mdfile.Close()
+
 	rawBytes, err := ioutil.ReadAll(mdfile)
 
 	// Get title from first line of file
 	lines := strings.Split(string(rawBytes), "\n")
 	title := strings.Replace(lines[0], "# ", "", -1)
 
-	mdfile.Close()
-
 	// Convert Markdown to HTML
 	html := blackfriday.Run(rawBytes)
 
 	// Create a Post struct
-	post := Post{
+	return &Post{
 		Title: title,
 		Body:  template.HTML(html),
 		Date:  date,
-		Slug:  slug}
-
-	return post
+		Slug:  slug,
+	}, nil
 }
 
-func createPostFile(post Post) {
+func createPostFile(post *Post) error {
 	// Create folder for HTML
 	newpath := filepath.Join("public/posts", post.Slug)
-	os.MkdirAll(newpath, os.ModePerm)
+	if err := os.MkdirAll(newpath, os.ModePerm); err != nil {
+		return err
+	}
 
 	// Create HTML file
 	f, err := os.Create("public/posts/" + post.Slug + "/" + "index.html")
 	if err != nil {
-		fmt.Println("Aaa!")
-		log.Fatal(err)
+		return err
 	}
+	defer f.Close()
 
 	// Generate final HTML file from template
-	t, _ := template.ParseFiles("post.html")
-	t.Execute(f, post)
-	f.Close()
+	t, err := template.ParseFiles("post.html")
+	if err != nil {
+		return err
+	}
+	return t.Execute(f, post)
 }
 
-func FnameWithoutExtension(fn string) string {
+func fnameWithoutExtension(fn string) string {
 	return strings.TrimSuffix(fn, path.Ext(fn))
 }
 
-func VerifyFilenameBaseFormat(f string) {
-	errorDescription := "I can't parse this filename. Make sure its name is formatted as: DD-MM-YYY-slug.md"
```

Comment:
While your version is indeed shorted and faster to read, first person was a deliberate choice for humane effect. I'd love it to stay that way.

Reply:
I just realized that:
- typo in "YYY"
- `DD-MM-YYYY` layout doesn't preserve calendar ordering of the files, so better layout would be `YYYY-MM-DD`